packages/amplify-graphql-transformer-core/src/transformation/transform.ts

```diff
@@ -172,13 +173,14 @@ export class GraphQLTransform {
    * @param schema The model schema.
    * @param references Any cloudformation references.
    */
-  public transform(schema: string): DeploymentResources {
+  public transform(schema: string, modelToDatasourceMap?: Map<string, DatasourceType>): DeploymentResources {
     this.seenTransformations = {};
     const parsedDocument = parse(schema);
     this.app = new App();
     const context = new TransformerContext(
       this.app,
       parsedDocument,
+      modelToDatasourceMap ?? new Map<string, DatasourceType>(),
```

Comment:
can we pass the input as-is, instead of this default initialization? I see that the target constructor accepts `undefined` - `modelToDatasourceMap: Map<string, DatasourceType> | undefined`

Reply:
The advantage is we can always use `context. modelToDatasourceMap` without doing a null check in transformer code. Ideally this should be a mandatory field on `TransformerContext` type. To fix some of the failing unit tests, I added `undefined`.

Reply:
But the transformer context still defines it as `public readonly modelToDatasourceMap: Map<string, DatasourceType> | undefined;`, so if we are accessing it, don't we still need to assert that it's defined?

Reply:
Removed the `undefined` on the type. Updated the unit tests to always pass a value.